### PR TITLE
feat(builder-api): delegations getter fn

### PR DESCRIPTION
**Pull request by namn-grg**
This PR adds a getter fn for delegations with test

### CODE REVIEW

**Comment by namn-grg** [comment]
Refer - https://github.com/chainbound/bolt-docs/pull/15

**Review by merklefruit — APPROVED** [review]
great job!

File: crates/api/src/builder/api.rs [hunk 1]
```diff
@@ -177,10 +179,10 @@ where
                     .map(|data| data.signed_constraints)
                     .collect::<Vec<SignedConstraints>>();
         
-                Ok(Json(constraints))
+                Ok(Json(constraints).into_response())
             }
             Ok(None) => {
-                Ok(Json(vec![])) // Return an empty vector if no constraints are found
+                Ok(StatusCode::NO_CONTENT.into_response())
```
Comment (merklefruit) [inline crates/api/src/builder/api.rs:185]: can you revert this back to an empty array? following the latest update in https://github.com/chainbound/bolt-docs/pull/15#discussion_r1777422621

same for delegations endpoint :D

Comment (namn-grg) [reply crates/api/src/builder/api.rs]: resolved

**Pull request merged**